modules/database/funcs.go

```diff
@@ -48,6 +48,17 @@ func NewDatabase(
 	}
 }
 
+// recieve a db pointer and remove it's finalizer
+// so it's set for deletion
+func Delete(
```

Comment:
not sure if we need a func for it. check how we do the finalizer remove in [1] for keystoneservice and keystoneendpoint. I think we could do the same for the mariadbdatabase finalizer.

[1] https://github.com/openstack-k8s-operators/placement-operator/blob/master/controllers/placementapi_controller.go#L210-L222